plugins/discourse-local-dates/test/javascripts/acceptance/download-calendar-with-ics-test.js

```diff
@@ -0,0 +1,118 @@
+import { click, visit } from "@ember/test-helpers";
+import { test } from "qunit";
+import { cloneJSON } from "discourse/lib/object";
+import { fixturesByUrl } from "discourse/tests/helpers/create-pretender";
+import { acceptance } from "discourse/tests/helpers/qunit-helpers";
+
+acceptance(
+  "Local Dates - Download calendar with embedded ICS data",
+  function (needs) {
+    needs.user({ "user_option.default_calendar": "none_selected" });
+    needs.settings({ discourse_local_dates_enabled: true });
+    needs.pretender((server, helper) => {
+      const response = cloneJSON(fixturesByUrl["/t/281.json"]);
+      const startDate = moment
+        .tz("America/Lima")
+        .add(1, "days")
+        .format("YYYY-MM-DD");
+
+      // Create sample ICS data
+      const icsData = `BEGIN:VCALENDAR
+VERSION:2.0
+PRODID:-//Discourse//EN
+BEGIN:VEVENT
+UID:test123
+DTSTAMP:20251201T000000Z
+DTSTART:20251201T130000Z
+DTEND:20251201T140000Z
+SUMMARY:Test Event from Calendar
+LOCATION:Test Location
+DESCRIPTION:Test Description
+END:VEVENT
+END:VCALENDAR`;
+
+      // Encode ICS data as base64url
+      const utf8Bytes = new TextEncoder().encode(icsData);
```

Comment:
Can't you use your `bbcodeAttributeEncode` stuff here?

Reply:
oh yikes I should also refactored this, will do